cpp/oneapi/dal/detail/error_reporting.cpp

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/detail/error_reporting.hpp"
+
+#define DEFINE_MESSAGE(id, text)                \
+    static const char* _##id = text;            \
```

Comment:
Is it global variable after macro substitution (for lines 27-32 of this file)? If so, could we avoid it (SDL requires to avoid if possible)? For instance, free function that returns const char * can be used instead?

Reply:
Actually it's not global variable, but global constant, but agree it's better to return directly from function.

Reply:
Not sure that do mean, but "constants" are "literals" in c++, and definitely it isn't literal. If you want to say that the variable type is const-qualified, it does not make sense for corresponding SDL task